sycl/include/syclcompat/memory.hpp

```diff
@@ -644,6 +644,9 @@ static inline void wait_and_free(void *ptr,
   }
 }
 
+// Anonymous namespace to disable ADL for functions which might clash (memcpy,
```

Comment:
Will it be better to use an inline namespace?

Reply:
Hey @ziranzha, as far as I can tell, an  inline namespace doesn't fix the ADL issue. Here's a [godbolt](https://godbolt.org/z/1h94YhG8W) with anonymous namespace. If we replace it with an inline namespace, [it breaks](https://godbolt.org/z/7Wffxr5Y3) with the "ambiguous" error. Please let me know if I've misunderstood your suggestion. Thanks!

Reply:
LGTM